src/utils/address.ts

```diff
@@ -16,7 +19,61 @@ import Network from '../models/network';
  *
  * @returns {string} output type of the address (p2pkh or p2sh)
  */
-export const getAddressType = (address: string, network: Network): string => {
+export function getAddressType(address: string, network: Network): 'p2pkh' | 'p2sh' {
   const addressObj = new Address(address, { network });
   return addressObj.getType();
 }
+
+export function deriveAddressFromXPubP2PKH(xpubkey: string, index: number, networkName: string): IAddressInfo {
```

Comment:
I am confused as to why we use the `networkName` actually, in my opinion the network object is better almost always but since we used the `networkName` in the old methods I tried to keep the behavior.